src/software/ai/play_selection_fsm.h

```diff
@@ -120,4 +133,16 @@ struct PlaySelectionFSM
    private:
     TbotsProto::AiConfig ai_config;
     std::shared_ptr<Play> current_play;
+
+    std::shared_ptr<BallPlacementPlay> ball_placement_play;
+    std::shared_ptr<EnemyBallPlacementPlay> enemy_ball_placement_play;
+    std::shared_ptr<EnemyFreekickPlay> enemy_free_kick_play;
+    std::shared_ptr<FreeKickPlay> free_kick_play;
+    std::shared_ptr<HaltPlay> halt_play;
+    std::shared_ptr<KickoffEnemyPlay> kickoff_enemy_play;
+    std::shared_ptr<KickoffFriendlyPlay> kickoff_friendly_play;
+    std::shared_ptr<OffensePlay> offense_play;
+    std::shared_ptr<PenaltyKickEnemyPlay> penalty_kick_enemy_play;
+    std::shared_ptr<PenaltyKickPlay> penalty_kick_play;
+    std::shared_ptr<StopPlay> stop_play;
```

Comment:
May be worth considering moving stuff from `Ai` to this class to get rid of these shared ptrs. This class can be renamed `PlaySelector` with methods for overriding the selected play and for returning primitives from the currently selected play

Reply:
In terms of semantics, wouldn't it make more intuitive sense for `Ai` to be returning `Primitives` instead of PlaySelector? Based on the name, `PlaySelector` wouldn't suggest that it does any robot control logic processing.

I do agree that we can probably move the override play logic into PlaySelector and merge the two selection paths. For that, I'll make a new issue for it #3099

Reply:
Actually nevermind, you're right. Now I see why we need shared ptrs (bc of the current_play ptr). Lets just leave it as is